### ISSUE TITLE
Unable to fetch org.scala-sbt:compiler-bridge_2.12:1.0.3

### ISSUE THREAD

**Issue by Abhi-joshi** [sbt/sbt#3684]
(See the guidelines for contributing, linked above)

## steps

Create a sample app with scala 2.12.4 and sbt 1.0.3
Do sbt run on it.

## problem
when you do a run on it sbt is generating an error 

```
[info] Attempting to fetch org.scala-sbt:compiler-bridge_2.12:1.0.3.
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: org.scala-sbt#compiler-bridge_2.12;1.0.3: 1
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[error] ## Exception when compiling 1 sources to G:\MyApplications\akka-quickstart-scala\target\scala-2.12\classes
[error] The compiler bridge sources org.scala-sbt:compiler-bridge_2.12:1.0.3:compile could not be retrieved.
[error]
[error]         Note: Unresolved dependencies path:
[error]                 org.scala-sbt:compiler-bridge_2.12:1.0.3
```

## expectation
Application will run smoothly with sbt 1.0.3

## notes

[issue_sbt-1.0.TXT](https://github.com/sbt/sbt/files/1423187/issue_sbt-1.0.TXT)

sbt version: *1.0.3*

**Comment by Abhi-joshi** [comment]
I am using windows 10 and sbt 1.0.2 (msi) is already installed on it. I don't know if there is conflict between installed version and build version in the specific project. @eed3si9n @dwijnand Is there anything which  am missing here?

**Comment by mkurz** [comment]
I have a a very similiiar, if not the same problem running sbt `1.0.3` on Ubuntu 17.04 with Java `9.0.1+11`.

Works fine with sbt `1.0.2`.

```
$ mkdir project
$ echo "sbt.version=1.0.3" > project/build.properties

$ sbt
[info] Loading project definition from /awesome-folder/awesome-project/project
[info] Updating {file:/awesome-folder/awesome-project/project/}awesome-project-build...
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.scala-lang#scala-library;2.12.4: Resolution failed several times for dependency: org.scala-lang#scala-library;2.12.4 {scala-tool=[default, optional(default)], provided=[default(compile)]}:: 
[warn] 	1
[warn] 	1
[warn] 	1
[warn] 	:: org.scala-sbt#sbt;1.0.3: Resolution failed several times for dependency: org.scala-sbt#sbt;1.0.3 {provided=[default(compile)]}:: 
[warn] 	1
[warn] 	1
[warn] 	1
[warn] 	:: org.scala-lang#scala-compiler;2.12.4: Resolution failed several times for dependency: org.scala-lang#scala-compiler;2.12.4 {scala-tool=[default, optional(default)]}:: 
[warn] 	1
[warn] 	1
[warn] 	1
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn] 	Note: Unresolved dependencies path:
[warn] 		org.scala-lang:scala-library:2.12.4 ((sbt.Classpaths.jvmBaseSettings) Defaults.scala#L2031)
[warn] 		  +- default:awesome-project-build:0.1-SNAPSHOT (scalaVersion=2.12, sbtVersion=1.0)
[warn] 		org.scala-sbt:sbt:1.0.3
[warn] 		  +- default:awesome-project-build:0.1-SNAPSHOT (scalaVersion=2.12, sbtVersion=1.0)
[warn] 		org.scala-lang:scala-compiler:2.12.4
[warn] 		  +- default:awesome-project-build:0.1-SNAPSHOT (scalaVersion=2.12, sbtVersion=1.0)
[error] sbt.librarymanagement.ResolveException: unresolved dependency: org.scala-lang#scala-library;2.12.4: Resolution failed several times for dependency: org.scala-lang#scala-library;2.12.4 {scala-tool=[default, optional(default)], provided=[default(compile)]}:: 
[error] 	1
[error] 	1
[error] 	1
[error] 
[error] unresolved dependency: org.scala-sbt#sbt;1.0.3: Resolution failed several times for dependency: org.scala-sbt#sbt;1.0.3 {provided=[default(compile)]}:: 
[error] 	1
[error] 	1
[error] 	1
[error] 
[error] unresolved dependency: org.scala-lang#scala-compiler;2.12.4: Resolution failed several times for dependency: org.scala-lang#scala-compiler;2.12.4 {scala-tool=[default, optional(default)]}:: 
[error] 	1
[error] 	1
[error] 	1
[error] 
[error] 	at sbt.internal.librarymanagement.IvyActions$.resolveAndRetrieve(IvyActions.scala:331)
[error] 	at sbt.internal.librarymanagement.IvyActions$.$anonfun$updateEither$1(IvyActions.scala:205)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.$anonfun$withModule$1(Ivy.scala:229)
[error] 	at sbt.internal.librarymanagement.IvySbt.$anonfun$withIvy$1(Ivy.scala:190)
[error] 	at sbt.internal.librarymanagement.IvySbt.sbt$internal$librarymanagement$IvySbt$$action$1(Ivy.scala:70)
[error] 	at sbt.internal.librarymanagement.IvySbt$$anon$3.call(Ivy.scala:77)
[error] 	at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:93)
[error] 	at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:78)
[error] 	at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:97)
[error] 	at xsbt.boot.Using$.withResource(Using.scala:10)
[error] 	at xsbt.boot.Using$.apply(Using.scala:9)
[error] 	at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:58)
[error] 	at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:48)
[error] 	at xsbt.boot.Locks$.apply0(Locks.scala:31)
[error] 	at xsbt.boot.Locks$.apply(Locks.scala:28)
[error] 	at sbt.internal.librarymanagement.IvySbt.withDefaultLogger(Ivy.scala:77)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:185)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:182)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.withModule(Ivy.scala:228)
[error] 	at sbt.internal.librarymanagement.IvyActions$.updateEither(IvyActions.scala:190)
[error] 	at sbt.librarymanagement.ivy.IvyDependencyResolution.update(IvyDependencyResolution.scala:20)
[error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[error] 	at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:38)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:91)
[error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:68)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$19(LibraryManagement.scala:104)
[error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:224)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:104)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:87)
[error] 	at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:149)
[error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:118)
[error] 	at sbt.Classpaths$.$anonfun$updateTask$5(Defaults.scala:2353)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:42)
[error] 	at sbt.std.Transform$$anon$4.work(System.scala:64)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:257)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error] 	at sbt.Execute.work(Execute.scala:266)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:257)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:167)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:32)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
[error] 	at java.base/java.lang.Thread.run(Thread.java:844)
[error] (*:update) sbt.librarymanagement.ResolveException: unresolved dependency: org.scala-lang#scala-library;2.12.4: Resolution failed several times for dependency: org.scala-lang#scala-library;2.12.4 {scala-tool=[default, optional(default)], provided=[default(compile)]}:: 
[error] 	1
[error] 	1
[error] 	1
[error] 
[error] unresolved dependency: org.scala-sbt#sbt;1.0.3: Resolution failed several times for dependency: org.scala-sbt#sbt;1.0.3 {provided=[default(compile)]}:: 
[error] 	1
[error] 	1
[error] 	1
[error] 
[error] unresolved dependency: org.scala-lang#scala-compiler;2.12.4: Resolution failed several times for dependency: org.scala-lang#scala-compiler;2.12.4 {scala-tool=[default, optional(default)]}:: 
[error] 	1
[error] 	1
[error] 	1
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
```

**Comment by xuwei-k** [comment]
I think sbt 1.0.3 dependency resolution does not work with Java 9( due to  
<del> https://github.com/sbt/librarymanagement/pull/173 or something? </del> https://github.com/sbt/ivy/pull/27 )

- https://travis-ci.org/xuwei-k/sbt-1.0.3-java-9/builds/294017814
- https://github.com/xuwei-k/sbt-1.0.3-java-9/commit/8c91f843d0c09c2bafa2dd0dc8c3775f00c3ecb7

![travis-image](https://user-images.githubusercontent.com/389787/32131799-38125564-bbf2-11e7-959f-cca3d5989d84.png)

**Comment by ScalaWilliam** [comment]
Having a very similar issue on Linux, 
```
$ java -version
java version "9.0.1"
Java(TM) SE Runtime Environment (build 9.0.1+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.1+11, mixed mode)
```

SBT 1.0.3

**Comment by xuwei-k** [comment]
https://github.com/sbt/ivy/pull/27

**Comment by dwijnand** [comment]
I'm going to assume that https://github.com/sbt/ivy/pull/27 fixes this issue.

**Comment by xuwei-k** [comment]
workaround

```diff
- sbt compile
+ sbt -Djava.specification.version=9.1 compile
```

**Comment by eed3si9n** [comment]
Fixed in sbt/ivy#27 and https://github.com/sbt/sbt/pull/3768.

**Comment by Deepak-nebhwani** [comment]
Please Add cross scala version into your build.sbt file. It worked in my case.

scalaVersion := "2.13.0"
crossScalaVersions := Seq("2.11.8","2.11.12", "2.12.4")
